pkg/operator/resource_reconciler.go

```diff
@@ -357,12 +361,22 @@ func (rr *ResourceReconciler) onStatefulSetDelete(ss *appsv1.StatefulSet) {
 
 // EnqueueForReconciliation asks for reconciling the object.
 func (rr *ResourceReconciler) EnqueueForReconciliation(obj metav1.Object) {
-	rr.reconcileQ.Add(obj.GetNamespace() + "/" + obj.GetName())
+	if rr.isOwnershipWrong(obj.GetLabels()) {
```

Comment:
It should work on annotations, not labels.

Reply:
done